core/tests/ledger_cleanup.rs

```diff
@@ -260,18 +270,28 @@ mod tests {
         let exit_cpu = Arc::new(AtomicBool::new(false));
         let sys = CpuStatsUpdater::new(&exit_cpu);
 
-        let mut generated_batches = VecDeque::<Vec<Shred>>::new();
+        let shreds: Arc<Mutex<VecDeque<Vec<Shred>>>> = Arc::new(Mutex::new(VecDeque::new()));
 
+        let mut entries_batch = make_many_slot_entries(0, batch_size, entries_per_slot).0;
         if pre_generate_data {
             let t0 = Instant::now();
             eprintln!("PRE_GENERATE_DATA: (this may take a while)");
             for i in 0..batches {
                 let start_slot = i * batch_size;
-                let (shreds, _) = make_many_slot_entries(start_slot, batch_size, entries_per_slot);
-                generated_batches.push_back(shreds);
+                let (new_shreds, _) = make_many_slot_entries(
+                    start_slot, batch_size, entries_per_slot);
+                shreds.lock().unwrap().push_back(new_shreds);
             }
             eprintln!("PRE_GENERATE_DATA: took {} ms", t0.elapsed().as_millis());
-        };
+        } else {
+            shreds.lock().unwrap().push_back(entries_batch.clone());
```

Comment:
nit:
- In the `pre_generate_case == true` case, we push in `batches` batches in the lines above.
- In the `pre_generate_case == false` case, we push this batch here, and then `batches` batches later for a total of `batches + 1` batches

Don't think we'd compare results between these cases anyways, but maybe don't push this batch in?

Reply:
>The logic was actually moved from the original line 356-363

Yep, I was suggesting that maybe this was unnecessary altogether

Reply:
Got it.  So in that case shall we just remove the original 356-363?  Since it looks to me we will end up having N+1 slots instead of N at the end.

Reply:
Think so on removing line 363. In regards to the following line (356 in old file)
```
let mut entries_batch = make_many_slot_entries(0, batch_size, entries_per_slot).0;
```
we still need this for `pre_generate_data == false` case. I'm good with keeping it where it is so that it is created outside of the timer. And, I think the `info!()` statement is good where you moved it to as well